marklogic/transactions.py

```diff
@@ -0,0 +1,100 @@
+
+logger = logging.getLogger(__name__)
+
+"""
+Defines classes to simplify usage of the REST endpoints defined at
+https://docs.marklogic.com/REST/client/transaction-management for managing transactions.
+"""
+
+
+class Transaction:
+    """
+    Represents a transaction created via
+    https://docs.marklogic.com/REST/POST/v1/transactions .
+
+    :param id: the ID of the new transaction, which is used for all subsequent
+    operations involving the transaction.
+    :param session: a requests Session object that is required for either committing or
+    rolling back the transaction, as well as for obtaining status of the transaction.
+    """
+
+    def __init__(self, id: str, session: Session):
+        self.id = id
+        self._session = session
+
+    def __enter__(self):
+        return self
+
+    """
+    Retrieve transaction status via
+    https://docs.marklogic.com/REST/GET/v1/transactions/[txid].
+    """
+
+    def get_status(self) -> dict:
+        return self._session.get(
+            f"/v1/transactions/{self.id}", headers={"Accept": "application/json"}
+        ).json()
+
+    """
+    Commits the transaction via
+    https://docs.marklogic.com/REST/POST/v1/transactions/[txid]. This is expected to be
+    invoked automatically via a Python context manager.
+    """
+
+    def commit(self) -> Response:
+        logger.debug(f"Committing transaction with ID: {self.id}")
+        return self._session.post(
+            f"/v1/transactions/{self.id}", params={"result": "commit"}
+        )
+
+    """
+    Rolls back the transaction via
+    https://docs.marklogic.com/REST/POST/v1/transactions/[txid]. This is expected to be
+    invoked automatically via a Python context manager.
+    """
+
+    def rollback(self) -> Response:
+        logger.debug(f"Rolling back transaction with ID: {self.id}")
+        return self._session.post(
+            f"/v1/transactions/{self.id}", params={"result": "rollback"}
+        )
+
+    def __exit__(self, *args):
+        response = (
+            self.rollback()
+            if len(args) > 1 and isinstance(args[1], Exception)
+            else self.commit()
+        )
+        assert (
+            204 == response.status_code
+        ), f"Could not end transaction; cause: {response.text}"
+
+
+class TransactionManager:
+    """
+    Simplifies creating a transaction via
+    https://docs.marklogic.com/REST/POST/v1/transactions .
+    """
+
+    def __init__(self, session: Session):
+        self._session = session
+
+    def create(self, name=None, time_limit=None, database=None) -> Transaction:
+        params = {}
+        if name:
+            params["name"] = name
+        if time_limit:
+            params["timeLimit"] = time_limit
+        if database:
+            params["database"] = database
+
+        response = self._session.post(
+            "/v1/transactions",
+            params=params,
+            headers={"Accept": "application/json"},
+        )
```

Comment:
Strangely, this doesn't match the documentation at https://docs.marklogic.com/REST/POST/v1/transactions

Reply:
Do you mean the use of the "Accept" header? I may have eagerly added that, I'll try removing it. 

I'm also going to change the code for when a transaction can't be committed / rolled back, seems like `raise_for_status()` should be used there.

Reply:
Also that the documentation says to use the "Location" header in the response (which I find strange).

Reply:
Added comments, the docs are incorrect.